test/spawn-options.node.js

```diff
@@ -0,0 +1,360 @@
+
+        ipfsd.stop(done)
+      })
+    })
+
+    // TODO ?? What is this test trying to prove?
+    describe('spawn a node and attach api', () => {
+      let ipfsd
+
+      it('create init and start node', function (done) {
+        this.timeout(20 * 1000)
+
+        df.spawn((err, _ipfsd) => {
+          expect(err).to.not.exist()
+          expect(_ipfsd).to.exist()
+          expect(_ipfsd.api).to.exist()
+          expect(_ipfsd.api.id).to.exist()
+
+          ipfsd = _ipfsd
+          done()
+        })
+      })
+
+      it('ipfsd.stop', function (done) {
+        this.timeout(20 * 1000)
+
+        ipfsd.stop(done)
+      })
+    })
+
+    describe('custom init options', () => {
+      it('custom config', function (done) {
+        this.timeout(20 * 1000)
+
+        const addr = '/ip4/127.0.0.1/tcp/5678'
+        const swarmAddr1 = '/ip4/127.0.0.1/tcp/35666'
+        const config = {
+          Addresses: {
+            Swarm: [
+              swarmAddr1
+            ],
+            API: addr
+          }
+        }
+
+        const options = { config: config }
+
+        waterfall([
+          (cb) => df.spawn(options, cb),
+          (ipfsd, cb) => ipfsd.getConfig('Addresses.API', (err, config) => {
+            expect(err).to.not.exist()
+            expect(config).to.eql(addr)
+            cb(null, ipfsd)
+          }),
+          (ipfsd, cb) => ipfsd.getConfig('Addresses.Swarm', (err, config) => {
+            expect(err).to.not.exist()
+
+            // TODO why would this not be always the same type?
+            if (typeof res === 'string') {
+              config = JSON.parse(config)
+            }
+            expect(config).to.eql([swarmAddr1])
+            cb(null, ipfsd)
+          })
+        ], (err, ipfsd) => {
+          expect(err).to.not.exist()
+          ipfsd.stop(done)
+        })
+      })
+    })
+
+    describe('custom repo path', () => {
+      // TODO why wouldn't this work from the browser if we use the
+      // remote endpoint?
+      if (!isNode) { return }
+
+      let ipfsd
+
+      it('allows passing custom repo path to spawn', function (done) {
+        this.timeout(50 * 1000)
+
+        const repoPath = tempDir(dfOpts)
+
+        const config = {
+          Addresses: {
+            Swarm: [
+              '/ip4/127.0.0.1/tcp/0/ws',
+              '/ip4/127.0.0.1/tcp/0'
+            ],
+            API: '/ip4/127.0.0.1/tcp/0',
+            Gateway: '/ip4/127.0.0.1/tcp/0'
+          }
+        }
+
+        const options = {
+          repoPath: repoPath,
+          config: config
+        }
+
+        series([
+          (cb) => df.spawn(options, (err, _ipfsd) => {
+            expect(err).to.not.exist()
+            ipfsd = _ipfsd
+            cb()
+          }),
+          (cb) => {
+            ipfsd.init(cb)
+          },
+          (cb) => {
+            ipfsd.start(cb)
+          }
+        ], (err) => {
+          expect(err).to.not.exist()
+          expect(fs.existsSync(repoPath)).to.be.ok()
```

Comment:
>   // TODO why wouldn't this work from the browser if we use the
     // remote endpoint?


because we use `fs.existsSync` - is there a way of making it work in the browser?